client/src/components/Tea/TeaList/TeaListContainer.jsx

```diff
@@ -4,23 +4,82 @@ import { deleteTea, getTeas } from "../../../actions/teaActions";
 import TeaList from "./TeaList";
 
 export class TeaListContainer extends React.Component {
+  state = {
+    sortedIDs: [],
+    memoizedIDs: {},
+    sortColumn: "",
+    sortOrder: ""
+  };
+
   handleDeleteClick = tea => {
     this.props.handleDelete(tea);
   };
 
+  handleSortClick = (columnName, sortOrder) => {
+    const list = this.props.teas.teaIDs;
+    let currentState = this.state.memoizedIDs;
+    let newSortOrder;
+
+    if (this.state.memoizedIDs[columnName]) {
+      newSortOrder = this.state.memoizedIDs[columnName][sortOrder];
+    } else {
+      let mapped = list.map((el, i) => {
+        return {
+          index: i,
+          value: this.props.teas.allTeas[el][columnName]
+        };
+      });
```

Comment:
This step seems unnecessary. Instead of creating this mapped list, why not just compare `this.props.teas.allTeas[a][columnName] > this.props.teas.allTeas[b][columnName]` inside your sort function? That means you wouldn't have to map it a second time to get the element IDs back.